client/src/components/QueryComponent/QueryComponent.js

```diff
@@ -65,6 +87,43 @@ class QueryComponent extends Component {
     this.setState({ selectedQuery: evt.target.value, selectedQueryKey: null });
   }
 
+  handleEditorChange = value => {
+    this.setState({ selectedQueryMultiline: value, selectedQueryKey: null });
+    let count = value.split("").filter(c => c === "\n").length;
+    count = count > 20 ? 20 : count;
+    this.setState({ rowHeight: `${(count + 1) * 19}px` });
+  }
+
+  handleEditorBlur = evt => {
+    this.setState({ displayEditor: false, selectedQuery: evt.target.value.replaceAll("\n", " ") });
+  }
+
+  handleEditorDidMount(editor, monaco) {
+    editor.focus();
+    const createDependencyProposals = (range, kind) => [ ...dependencyProposalSegmentOne(range, kind), ...dependencyProposalSegmentTwo(range, kind), ...dependencyProposalSegmentThree(range, kind) ];
+    monaco.languages.registerCompletionItemProvider("sql", {
+      provideCompletionItems: (model, position) => {
+        const word = model.getWordUntilPosition(position);
+        const range = {
+          startLineNumber: position.lineNumber,
+          endLineNumber: position.lineNumber,
+          startColumn: word.startColumn,
+          endColumn: word.endColumn
+        };
+        return {
+          suggestions: createDependencyProposals(range, monaco.languages.CompletionItemKind.Function)
+        };
+      }
+    });
+  }
+
+  handleEditorDidUnMount(editor, monaco) {
+    editor.unFocus();
+    monaco.languages.registerCompletionItemProvider("", {
+      provideCompletionItems: () => null
+    }).dispose();
```

Comment:
I think we need to call dispose on the *original* return value of `registerCompletionItemProvider` (i.e. from handleEditorDidMount) rather than calling it again here? Or is there documentation to the contrary?

Reply:
It is actually not working for the intention I need it. Ill just remove it.